walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -93,15 +118,18 @@ else if (event.getWhen().end() > slot.start()){
   }
 
   /**Returns true if at least one attendee in Meeting Request is an attendee in the Event */
+  // Runtime: addAll and retainAll are O(n) time (I could be wrong on that). 
   private boolean overlappingAttendees(MeetingRequest request, Event event){
-      for (String attendee : event.getAttendees()){
-          if (request.getAttendees().contains(attendee))
-            return true;
-        }
-      return false;
+      // Since request.getAttendees returns an unmuttable set. Need to copy it.
+      Collection<String> requestedAttendees = new ArrayList<>();
+      requestedAttendees.addAll(request.getAttendees());
```

Comment:
You can also just use the copy constructor:

```java
ArrayList<String> requestedAttendees = new ArrayList<>(request.getAttendees());
```

(I didn't catch that `#retainAll` mutated the collection in place. My mistake.)